src/Libraries/CoreNodeModels/DropDown.cs

```diff
@@ -175,6 +175,52 @@ protected static string XmlUnescape(string escaped)
             return node.InnerText;
         }
 
-        public abstract void PopulateItems();
+        public void PopulateItems()
+        {
+            var currentSelection = string.Empty;
+            if (items.Count != 0 && SelectedIndex > -1)
+            {
+                currentSelection = items.ElementAt(SelectedIndex).Name;
+            }
+            var selectionState = PopulateItemsCore(currentSelection);
+            if (selectionState == SelectionState.Restore)
+            {
+                for (int i = 0; i < items.Count; i++)
```

Comment:
Should we set `SelectedIndex = -1` before the `for` loop just in case a match cannot be found?